### ISSUE TITLE
Stray parent?

### ISSUE THREAD

**Issue by apowers313** [hildjj/node-cbor#64]
I'm wondering if I found a bug with zero-length maps where the output includes a very complex object, or if the example I show below is actually the correct behavior and I just don't understand what the parser is returning.

I have an CBOR object that looks like this:

<pre>
attestationObject
A3
    63
        66 6D 74                    // "fmt"
    64
        6E 6F 6E 65                 // "none"
    <b>67
        61 74 74 53 74 6D 74        // "attStmt"
    A0                              // map(0)</b>
    68
        61 75 74 68 44 61 74 61     // "authData"
    59 01 26
        49 96 0D E5 88 0E 8C 68 74 34 17 0F 64 76 60 5B
        [ ... snip ... ]
</pre>

What I get out of the parser looks like this:

<pre>
{
  fmt: 'none',
  <b>attStmt: {
    [Symbol(parent)]: ['fmt',
      'none',
      'attStmt', [Circular],
      'authData', <
      Buffer 49 96 0 d e5 88 0e 8 c 68 74 34 17 0 f 64 76 60 5 b 8 f e4 ae b9 a2 86 32 c7 99 5 c f3 ba 83 1 d 97 63 41 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00... > , 
      [Symbol(parent)]: null, 
      [Symbol(major type)]: 5
    ]
  },</b>
  authData: < Buffer 49 96 0 d e5 88 0e 8 c 68 74 34 17 0 f 64 76 60 5 b 8 f e4 ae b9 a2 86 32 c7 99 5 c f3 ba 83 1 d 97 63 41 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00... >
}
</pre>

The map key `attStmt` has a value that is a zero-length map. I don't understand why the object returned by the parser has a `parent` value. Is there something I'm missing?

**Comment by hildjj** [comment]
Are you using the latest version?  I've fixed bugs in this space in the last year.  With v4.0.0, and the following hex-encoded input:

`A363666D74646E6F6E656761747453746D74A068617574684461746159001049960DE5880E8C687434170F6476605B`

(edited slightly from your example to set the length of the last field to the data you provided)
I get the following object created:

```
{ fmt: 'none',
    attStmt: {},
    authData: <Buffer 49 96 0d e5 88 0e 8c 68 74 34 17 0f 64 76 60 5b> }
```

**Comment by apowers313** [comment]
So sorry, my dependency was pinned to 3.0. You're right, this is fixed in 4.0.
